exec-healthz/Makefile

```diff
@@ -53,11 +53,11 @@ endif
 	cd $(TEMP_DIR) && sed -i "s|BASEIMAGE|$(BASEIMAGE)|g" Dockerfile
 
 	# Compile the binary inside a container for reliable builds
-	docker run --rm -it -v $(TEMP_DIR):/build golang:$(GOLANG_VERSION) /bin/bash -c "make -C /build server ARCH=$(ARCH)"
+	docker run --rm -it -v $(TEMP_DIR):/build golang:$(GOLANG_VERSION) /bin/bash -c "mv /build/vendor/* /go/src && make -C /build server ARCH=$(ARCH) TAG=$(TAG)"
```

Comment:
The previous command tried to build under `/build`, which is outside $GOPATH and $GOROOT. So go will not look in the vendor folder(golang/go#12511).

What about this command?
`docker run --rm -it -v $(TEMP_DIR):/go/src/build golang:$(GOLANG_VERSION) /bin/bash -c "make -C /go/src/build server ARCH=$(ARCH) TAG=$(TAG)"`